recipe/test_cmake/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.17.0 FATAL_ERROR)
+
+project(cmake_test_cccl LANGUAGES CUDA)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
+find_package(CUDAToolkit REQUIRED)
+find_package(CCCL REQUIRED)
```

Comment:
We want to validate that the CCCL includes are coming from the package we just made and not the version packaged in the CUDAToolkit. 
I believe that the following check should be added at the end:
```cmake
get_target_property(cccl_raw_includes CCCL::libcudacxx INTERFACE_INCLUDE_DIRECTORIES)
foreach(item IN LISTS cccl_raw_includes)
  foreach(toolkit_path IN LISTS CUDAToolkit_INCLUDE_DIRS)
    cmake_path(IS_PREFIX toolkit_path "${item}" NORMALIZE is_prefix)
    if(is_prefix)
      message(FATAL_ERROR "Failed to find conda version of the CCCL. Instead found version within the CUDAToolkit")
    endif()
  endforeach()
endforeach()
```

Reply:
Right now this is run on CUDA 11.8 where the NVIDIA CUDA-based Docker images are used. Here are the results we see with that

```
-- The CUDA compiler identification is NVIDIA 11.8.89
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: $PREFIX/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Found CUDAToolkit: /usr/local/cuda/targets/x86_64-linux/include (found version "11.8.89") 
-- Found libcudacxx: $PREFIX/lib/cmake/libcudacxx/libcudacxx-config.cmake (found suitable version "2.2.0.0", minimum required is "2.2.0.0") 
-- Found Thrust: $PREFIX/lib/cmake/thrust/thrust-config.cmake (found suitable exact version "2.2.0.0") 
-- Found CUB: $PREFIX/lib/cmake/cub/cub-config.cmake (found suitable version "2.2.0.0", minimum required is "2.2.0.0") 
-- Found CCCL: $PREFIX/lib/cmake/cccl/cccl-config.cmake (found version "2.2.0.0") 
-- CCCL info:
--     config: $PREFIX/lib/cmake/cccl/cccl-config.cmake
--     version: 2.2.0.0
-- Configuring done (3.3s)
-- Generating done (0.0s)
-- Build files have been written to: $SRC_DIR/test_cmake
```

So that seems to point `$PREFIX` as expected

On CUDA 12, we ensured that `$PREFIX` (so this `cccl`) come [before the `targets` directory in the search order]( https://github.com/conda-forge/cuda-nvcc-feedstock/blob/bb6f112003bc990193e31788b68d222f32e3129d/recipe/activate.sh#L22 )

If we are concerned about `cuda-cccl` from `targets` being picked up, should we just override `CMAKE_FIND_ROOT_PATH` in the test?